tests/TestCase/Controller/JsonApiIntegrationTest.php

```diff
@@ -0,0 +1,51 @@
+<?php
+namespace App\Test\TestCase\Controller;
+
+use Crud\TestSuite\IntegrationTestCase;
+
+class JsonApiIntegrationTest extends IntegrationTestCase
+{
+    public $fixtures = [
+        'plugin.crud.countries'
+    ];
+
+    /**
+     * Test GET request
+     *
+     * @return void
+     */
+    public function testGet()
+    {
+        $this->configRequest([
+            'headers' => ['Accept' => 'application/vnd.api+json']
+        ]);
+        $result = $this->get('/countries');
+
+        $this->assertResponseOk();
+        $this->assertResponseCode(200);
+        $this->assertResponseNotEmpty();
+    }
+
+    /**
+     * Test failing POST request due to validation errors, should return status code 422.
+     *
+     * @return void
+     */
+    public function testPostValidationError()
+    {
+        $this->configRequest([
+            'headers' => [
+                'Accept' => 'application/vnd.api+json',
+                'Content-Type' => 'application/vnd.api+json'
+            ]
+        ]);
+        $result = $this->post('/countries', []);
+
+        // These tests do not pass, the expected JSON API validation exception is not thrown
+         $this->assertResponseError();
+         $this->assertResponseCode(422);
+
+        // and this prints the index JSON API instead ?!?
```

Comment:
It does so because you haven't connected any RESTful route. You should do so in `setUp()`.